src/EnergyPlus/WindowManager.hh

```diff
@@ -388,213 +306,202 @@ namespace WindowManager {
+                                                    //  (program calculates from tir and emis)
+    std::array<Real64, 10> AbsRadGlassFace = {0.0}; // Solar radiation and IR radiation from internal gains absorbed by glass face
+    std::array<Real64, 10> thetas = {0.0};          // Glass surface temperatures (K)
+    std::array<Real64, 10> thetasPrev = {0.0};      // Previous-iteration glass surface temperatures (K)
+    std::array<Real64, 10> fvec = {0.0};            // Glass face heat balance function
+
+    std::array<Real64, 5> hrgap = {0.0}; // Radiative gap conductance
+
+    Real64 A23P = 0.0; // Intermediate variables in glass face
+    Real64 A32P = 0.0;
+    Real64 A45P = 0.0;
+    Real64 A54P = 0.0;
+    Real64 A67P = 0.0;
+    Real64 A76P = 0.0;
+    Real64 A23 = 0.0; // heat balance equations
+    Real64 A45 = 0.0;
+    Real64 A67 = 0.0;
+
+    static int constexpr MaxNumOfIncidentAngles = 20;
+    static int constexpr MaxSpectralDataElements = 800; // Maximum number in Spectral Data arrays.
     // TEMP MOVED FROM DataHeatBalance.hh -BLB
 
-    Array2D<Real64> wlt;      // Spectral data wavelengths for each glass layer in a glazing system
-                              // Following data, Spectral data for each layer for each wavelength in wlt
-    Array2D<Real64> t;        // normal transmittance
-    Array2D<Real64> rff;      // normal front reflectance
-    Array2D<Real64> rbb;      // normal back reflectance
-    Array2D<Real64> tPhi;     // transmittance at angle of incidence
-    Array2D<Real64> rfPhi;    // front reflectance at angle of incidence
-    Array2D<Real64> rbPhi;    // back reflectance at angle of incidence
-    Array2D<Real64> tadjPhi;  // transmittance at angle of incidence
-    Array2D<Real64> rfadjPhi; // front reflectance at angle of incidence
-    Array2D<Real64> rbadjPhi; // back reflectance at angle of incidence
-
-    Array1D_int numpt;              // Number of spectral data wavelengths for each layer; =2 if no spectra data for a layer
-    Array1D<Real64> stPhi;          // Glazing system transmittance at angle of incidence for each wavelength in wle
-    Array1D<Real64> srfPhi;         // Glazing system front reflectance at angle of incidence for each wavelength in wle
-    Array1D<Real64> srbPhi;         // Glazing system back reflectance at angle of incidence for each wavelenth in wle
-    Array2D<Real64> saPhi;          // For each layer, glazing system absorptance at angle of incidence
-                                    // for each wavelenth in wle
-    Array2D<Real64> top;            // Transmittance matrix for subr. op
-    Array2D<Real64> rfop;           // Front reflectance matrix for subr. op
-    Array2D<Real64> rbop;           // Back transmittance matrix for subr. op
+    std::array<std::array<Real64, MaxSpectralDataElements>, 5> wlt = {0.0}; // Spectral data wavelengths for each glass layer in a glazing system
```

Comment:
Probably better off implementing this as an array of array of structs than as seven separate arrays of arrays. I am coming over to this side more and more.